### PR TITLE
Contributing Glucose 4.1

### PR DESCRIPTION
**Pull request by danielleberre**
We would like to contribute Glucose 4.1 to Alloy.

We include the source code, the macos and linux binaries (both 32 and 64 arch). The windows binaries are missing.

We are not quite sure which files should be added to the repository (e.g. generated files, gradle config files, etc).
 
Guidance on that topic would be welcome :)

### CODE REVIEW

File: gradlew [hunk 1]
```diff
@@ -28,16 +44,16 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (aleksandarmilicevic) [inline gradlew]: any reason for changing this?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
```
Comment (aleksandarmilicevic) [inline gradle/wrapper/gradle-wrapper.properties]: why do we need to check in the `gradle-wrapper.jar` binary file?

File: org.alloytools.alloy.application/.classpath [hunk 3]
```diff
@@ -3,7 +3,15 @@
...
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">
+		<attributes>
+			<attribute name="module" value="true"/>
+		</attributes>
+	</classpathentry>
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/org.alloytools.alloy.core"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/org.alloytools.kodkod.core"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/3"/>
+	<classpathentry kind="lib" path="/home/lonca/projects/org.alloytools.alloy/org.alloytools.alloy.application/lib/apple-osx-ui.jar"/>
```
Comment (aleksandarmilicevic) [inline org.alloytools.alloy.application/.classpath]: `/home/lonca` is likely not going to work for everyone

File: org.alloytools.alloy.core/.classpath [hunk 4]
```diff
@@ -2,9 +2,15 @@
...
+	<classpathentry kind="src" path="src/main/resources"/>
+	<classpathentry kind="src" path="src/test/resources"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">
+		<attributes>
+			<attribute name="module" value="true"/>
+		</attributes>
+	</classpathentry>
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/org.alloytools.kodkod.core"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/4"/>
```
Comment (aleksandarmilicevic) [inline org.alloytools.alloy.core/.classpath]: Are any of these changes essential to your main contribution?  If they are not, please revert them.

File: org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/parser/CompLexer.java [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/* The following code was generated by JFlex 1.4.1 on 4/8/18 9:54 AM */
+/* The following code was generated by JFlex 1.4.1 on 04/06/2019 10:43 */
```
Comment (aleksandarmilicevic) [inline org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/parser/CompLexer.java]: Did you actually change the lexer?  If you didn't, please revert this file

File: org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/parser/CompSym.java [hunk 6]
```diff
@@ -1,7 +1,7 @@
 
 //----------------------------------------------------
 // The following code was generated by CUP v0.11a beta 20060608
-// Sun Apr 08 09:54:30 PDT 2018
+// Tue Jun 04 10:43:46 CEST 2019
```
Comment (aleksandarmilicevic) [inline org.alloytools.alloy.core/src/main/java/edu/mit/csail/sdg/parser/CompSym.java]: please revert

File: org.alloytools.kodkod.core/.classpath [hunk 7]
```diff
@@ -2,7 +2,14 @@
...
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8">
+		<attributes>
+			<attribute name="module" value="true"/>
+		</attributes>
+	</classpathentry>
 	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry kind="lib" path="/home/lonca/projects/org.alloytools.alloy/org.alloytools.kodkod.nativesat/lib/sat4j-core-v20120709/org.sat4j.core.jar"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/3"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/org.alloytools.kodkod.nativesat"/>
```
Comment (aleksandarmilicevic) [inline org.alloytools.kodkod.core/.classpath]: same question as before.

File: org.alloytools.kodkod.core/.settings/org.eclipse.jdt.core.prefs [hunk 8]
```diff
@@ -21,8 +21,11 @@ org.eclipse.jdt.core.codeComplete.staticFinalFieldSuffixes=
 org.eclipse.jdt.core.compiler.annotation.inheritNullAnnotations=disabled
 org.eclipse.jdt.core.compiler.annotation.missingNonNullByDefaultAnnotation=ignore
 org.eclipse.jdt.core.compiler.annotation.nonnull=org.eclipse.jdt.annotation.NonNull
+org.eclipse.jdt.core.compiler.annotation.nonnull.secondary=
```
Comment (aleksandarmilicevic) [inline org.alloytools.kodkod.core/.settings/org.eclipse.jdt.core.prefs]: Please don't gratuitously change the preferences in this PR.

If you want to suggest better preferences, please open a new PR.

File: org.alloytools.kodkod.nativesat/lib/glucose-syrup-4.1/build.sh [hunk 9]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+cd sources/simp
+make rs
+cp glucose_static ../..
```
Comment (aleksandarmilicevic) [inline org.alloytools.kodkod.nativesat/lib/glucose-syrup-4.1/build.sh]: what's this?

**Comment by danielleberre** [comment]
Hi Aleksandar,

Thanks for your feedback.

We did our best to fix all those issues.

**Comment by danielleberre** [comment]
ok, thanks.

**Comment by pkriens** [comment]
LGTM so far. I need to rework my stale PR so then I will address any more issues.

I am ok to merge

**Pull request merged**